packages/fast-colors/src/colorlib/parseColor.ts

```diff
@@ -0,0 +1,219 @@
+    moccasin: new ColorRGBA64(1, 0.894118, 0.709804, 1),
+    navajowhite: new ColorRGBA64(1, 0.870588, 0.678431, 1),
+    navy: new ColorRGBA64(0, 0, 0.501961, 1),
+    oldlace: new ColorRGBA64(0.992157, 0.960784, 0.901961, 1),
+    olive: new ColorRGBA64(0.501961, 0.501961, 0, 1),
+    olivedrab: new ColorRGBA64(0.419608, 0.556863, 0.137255, 1),
+    orange: new ColorRGBA64(1, 0.647059, 0, 1),
+    orangered: new ColorRGBA64(1, 0.270588, 0, 1),
+    orchid: new ColorRGBA64(0.854902, 0.439216, 0.839216, 1),
+    palegoldenrod: new ColorRGBA64(0.933333, 0.909804, 0.666667, 1),
+    palegreen: new ColorRGBA64(0.596078, 0.984314, 0.596078, 1),
+    paleturquoise: new ColorRGBA64(0.686275, 0.933333, 0.933333, 1),
+    palevioletred: new ColorRGBA64(0.858824, 0.439216, 0.576471, 1),
+    papayawhip: new ColorRGBA64(1, 0.937255, 0.835294, 1),
+    peachpuff: new ColorRGBA64(1, 0.854902, 0.72549, 1),
+    peru: new ColorRGBA64(0.803922, 0.521569, 0.247059, 1),
+    pink: new ColorRGBA64(1, 0.752941, 0.796078, 1),
+    plum: new ColorRGBA64(0.866667, 0.627451, 0.866667, 1),
+    powderblue: new ColorRGBA64(0.690196, 0.878431, 0.901961, 1),
+    purple: new ColorRGBA64(0.501961, 0, 0.501961, 1),
+    red: new ColorRGBA64(1, 0, 0, 1),
+    rosybrown: new ColorRGBA64(0.737255, 0.560784, 0.560784, 1),
+    royalblue: new ColorRGBA64(0.254902, 0.411765, 0.882353, 1),
+    saddlebrown: new ColorRGBA64(0.545098, 0.270588, 0.07451, 1),
+    salmon: new ColorRGBA64(0.980392, 0.501961, 0.447059, 1),
+    sandybrown: new ColorRGBA64(0.956863, 0.643137, 0.376471, 1),
+    seagreen: new ColorRGBA64(0.180392, 0.545098, 0.341176, 1),
+    seashell: new ColorRGBA64(1, 0.960784, 0.933333, 1),
+    sienna: new ColorRGBA64(0.627451, 0.321569, 0.176471, 1),
+    silver: new ColorRGBA64(0.752941, 0.752941, 0.752941, 1),
+    skyblue: new ColorRGBA64(0.529412, 0.807843, 0.921569, 1),
+    slateblue: new ColorRGBA64(0.415686, 0.352941, 0.803922, 1),
+    slategray: new ColorRGBA64(0.439216, 0.501961, 0.564706, 1),
+    slategrey: new ColorRGBA64(0.439216, 0.501961, 0.564706, 1),
+    snow: new ColorRGBA64(1, 0.980392, 0.980392, 1),
+    springgreen: new ColorRGBA64(0, 1, 0.498039, 1),
+    steelblue: new ColorRGBA64(0.27451, 0.509804, 0.705882, 1),
+    tan: new ColorRGBA64(0.823529, 0.705882, 0.54902, 1),
+    teal: new ColorRGBA64(0, 0.501961, 0.501961, 1),
+    thistle: new ColorRGBA64(0.847059, 0.74902, 0.847059, 1),
+    tomato: new ColorRGBA64(1, 0.388235, 0.278431, 1),
+    transparent: new ColorRGBA64(0, 0, 0, 0),
+    turquoise: new ColorRGBA64(0.25098, 0.878431, 0.815686, 1),
+    violet: new ColorRGBA64(0.933333, 0.509804, 0.933333, 1),
+    wheat: new ColorRGBA64(0.960784, 0.870588, 0.701961, 1),
+    white: new ColorRGBA64(1, 1, 1, 1),
+    whitesmoke: new ColorRGBA64(0.960784, 0.960784, 0.960784, 1),
+    yellow: new ColorRGBA64(1, 1, 0, 1),
+    yellowgreen: new ColorRGBA64(0.603922, 0.803922, 0.196078, 1),
+};
+
+// accepts formats
+// #RRGGBB

```

Comment:
I think we should split up the parsers into smaller functions, with `parseColor` invoking those functions as necessary:
`parseRGB`, `parseRGBA`, and `parseHEX` or something similar.

I expect almost every consumer of this library to depend on color parsing - however I think a much less likely scenario is a user needs to parse the color "yellow" (or any other named color). As this is architected today, I cannot depend on that function without *also* depending on the 147 named colors above.

Splitting out these functions would give app authors the proper tools to remove code they don't need.